src/Library/types.ts

```diff
@@ -0,0 +1,85 @@
+export type Author = {
+    readonly name: string;
+    readonly url: string;
+    readonly position: string;
+};
+
+enum DefaultCategories {
+    All = 'All',
+    Popular = 'Popular'
+}
+
+enum PartnersCategories {
+    Reastream = 'Reastream'
+}
+
+export type LibraryCategories = DefaultCategories | PartnersCategories;
+
+export enum LibraryDirections {
+    All = 'All',
+    Graphics = 'Graphics',
+    Dev = 'Dev',
+    Marketing = 'Marketing',
+    Management = 'Management',
+    Advertising = 'Advertising'
+}
+
+export enum LibraryFormats {
+    All = 'All',
+    Conference = 'Conference',
+    Video = 'Video',
+    Tutorial = 'Tutorial',
+    Interview = 'Interview'
+}
+
+export enum LibraryLevels {
+    All = 'All',
+    Beginner = 'Beginner',
+    Specialist = 'Specialist'
+}
+
+enum VideoSize {
+    Default = 'Default',
+    Featured = 'Featured'
+}
+
+export interface Video {
+    readonly id: number;
+    readonly title: string;
+    readonly description: string;
+    readonly mainCategory?: LibraryCategories;
+    readonly categories: LibraryCategories[];
+    readonly publishDate?: string;
+    readonly authors: Author[];
+    readonly preview: string;
+    readonly source?: string;
+    readonly isFavorite: boolean;
+    readonly isPrivate: boolean;
+    readonly videoSize: VideoSize;
+}
+
+export type VideoСompilation = {
+    readonly title: string;
+    readonly link: string;
+    readonly list: Video[] | null;
+};
+
+export type RequestParams = {
+    readonly query: string;
+    readonly direction: LibraryDirections;
```

Comment:
The same here, it might be better to use singulars

```
readonly direction: LibraryDirection;
readonly format: LibraryFormat;
readonly level: LibraryLevel;
readonly category: LibraryCategory;
```

What do u think?

Reply:
Good point, thanks.